oss-internship-2020/libtiff/test/helper.cc

```diff
@@ -0,0 +1,37 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "helper.h"
+
+static auto* g_in_dir = new std::string();
+
+std::string GetImagesDir() {
+  std::string cwd = sandbox2::file_util::fileops::GetCWD();
+  auto find = cwd.rfind("/build");
+  if (find == std::string::npos) {
+    LOG(ERROR) << "Something went wrong: CWD don't contain build dir. "
+               << "Please run tests from build dir, path might be incorrect\n";
+
+    return cwd + "/test/images";
+  }
+
+  return cwd.substr(0, find) + "/test/images";
+}
+
+std::string GetFilePath(const std::string& filename) {
+  if (g_in_dir->empty()) {
```

Comment:
Are you using `g_in_dir` anywhere else but here? You can put it in there otherwise, mark it as static and directly initialize it using `GetImagesDir()`.

Reply:
I only use this here.

Reply:
Then please put the variable inside this function and initialize it with nullptr. Then when it is called check for nullptr and initialize it.